crates/ty_python_semantic/src/types/call/bind.rs

```diff
@@ -1412,7 +1412,7 @@ impl<'db> Binding<'db> {
         }
 
         num_synthetic_args = 0;
-        for (argument_index, (argument, argument_type)) in argument_types.iter().enumerate() {
+        for (argument_index, (argument, mut argument_type)) in argument_types.iter().enumerate() {
```

Comment:
Does this need to be mutated? It seems like in the loop, `argument_type = argument_type.apply_specialization(...)` indicates that it might not be required.